grpc/src/compiler/cpp_generator.cc

```diff
@@ -97,7 +97,7 @@ void PrintIncludes(grpc_generator::Printer *printer,
     }
   }
 
-  for (auto i = headers.begin(); i != headers.end(); i++) {
```

Comment:
This file is shared with the GRPC project, so they won't be retained if not also fixed upstream.

Reply:
Indeed, these files are copies from GRPC project.
I examined the `java_generator.cc`. Always true condition `if (!interface)` does not affect the correctness of the generated code.
I have removed these changes from PR.